corehq/ex-submodules/pillowtop/pillow/interface.py

```diff
@@ -374,13 +374,22 @@ def __record_change_metric_in_datadog(self, metric, change, processing_time=None
             metrics_counter(metric, tags=metric_tags)
 
             change_lag = (datetime.utcnow() - change.metadata.publish_timestamp).total_seconds()
-            metrics_gauge('commcare.change_feed.change_lag', change_lag, tags={
-                'pillow_name': self.get_name(),
-                'topic': _topic_for_ddog(
-                    TopicPartition(change.topic, change.partition)
-                    if change.partition is not None else change.topic
-                ),
-            }, multiprocess_mode=MPM_MAX)
+            dd_topic = _topic_for_ddog(
+                TopicPartition(change.topic, change.partition)
+                if change.partition is not None else change.topic)
+            metrics_gauge(
+                'commcare.change_feed.change_lag',
+                change_lag,
+                tags={
+                    'pillow_name': self.get_name(),
+                    'topic': dd_topic
+                },
+                multiprocess_mode=MPM_MAX
+            )
+            if change.partition:
```

Comment:
What is the reason for checking `change.partition` here? It isn't used in the block. I'd assume maybe `dd_topic`?

Reply:
`dd_topic` is created using topic and partition, if there is not partition it just uses topic. But for the case pillow we need to have both the values. This is just an extended check which may or maynot be useful. I added it to be on safer side.

Reply:
Ok makes sense. I think it could be worth adding a comment since it isn't clear in code why you would need a partition to report usage to the pillow lag gauge. What do you think?

Reply:
https://github.com/dimagi/commcare-hq/pull/34800/commits/75bb94a807a847b2290806d4cd3b24d7106e79bc